backend/msgpack/writer.go

```diff
@@ -209,6 +215,33 @@ func (w *writer) encodeWithLock(
 				PoliciesList: pl,
 			}
 			err = encoder.EncodeBatchTimerWithPoliciesList(btp)
+			if err != nil {
+				break
+			}
+
+			// If the buffer isn't big enough continue to the next iteration.
+			if buffer.Len() < w.flushSize {
```

Comment:
super nit: `if sizeAfter := buffer.Len(); sizeAfter < w.flushSize` is slightly easier to read.